src/main/java/com/google/sps/servlets/getVideosServlet.java

```diff
@@ -79,46 +80,64 @@ public void doPut (HttpServletRequest request, HttpServletResponse response) thr
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     GetServletsUtility.deleteResultsOfQueryFromDatastore(query, datastore,KIND);
 
-    // Get string from api.
+    // Map to store strings and catagory types
+    HashMap<String, String> videoMap = new HashMap<String, String>();
+
+    // Get yoga video string from api.
     String youtubeapiKey = getSecretKey.accessSecretVersion("298755462", "youtube_api_key", "2"); // Get hidden api key from gcloud secret manager.
-    URL url = new URL(baseURL + youtubeapiKey);
-    StringBuilder strBuf = new StringBuilder();  
-    String videos = getStringFromAPI.getStringFromAPIMethod(url, strBuf, null, null, logger, request, response);
-    if (videos.equals("end")) return; // Returns if exception caught.
-
-    // Put data from api string into database.
-    JSONObject obj = new JSONObject(videos);
-    JSONArray videoData = obj.getJSONArray("items");
-    int numVideos = videoData.length();
-    
-    String contentDetailsURL = "https://www.googleapis.com/youtube/v3/videos?part=contentDetails&id=";
-    String videoLinkBaseURL = "https://www.youtube.com/watch?v=";
-
-    for (int i = 0; i < numVideos; ++i) {
-      JSONObject currentVideo = videoData.getJSONObject(i);
-      Entity videoEntity = new Entity(KIND);
-
-      // Get video details from api.
-      String videoId = currentVideo.getJSONObject("id").getString("videoId"); // Get videoId.
-      URL getContentDetailsURL = new URL(contentDetailsURL + videoId + "&key=" + youtubeapiKey);
-      StringBuilder videoStrBuf = new StringBuilder();  
-      String videoDetails = getStringFromAPI.getStringFromAPIMethod(getContentDetailsURL, videoStrBuf, null, null, logger, request, response);
-      if (videos.equals("end")) return; // Returns if exception caught.
-
-      // Get duration from parsed string from api.
-      JSONObject videoDetailObj = new JSONObject(videoDetails);
-      String duration = videoDetailObj.getJSONArray("items").getJSONObject(0).getJSONObject("contentDetails").getString("duration");
-
-      // Convert duration string into seconds.
-      long durationInSeconds = Duration.parse(duration).getSeconds();
+    URL yogaUrl = new URL(baseURL + "yoga" + "&type=video&key=" + youtubeapiKey); 
```

Comment:
The code for yoga, workout and meditation are highly similar. Could you put them into a shared method?

You can use yoga, workout and meditation as a parameter.

Reply:
Yes sure! Will do that now.